docs/faqs.md

```diff
@@ -0,0 +1,145 @@
+two major reason why they might make less sense going forward:
+
+- We are only just beginning to scale quantum software to FTQC regimes, so have not experienced the
+  performance issues associated with (re-)parsing human readable formats, compared to formats
+  designed for machine-readability.
+
+- the tooling around IR and exchange formats has not been mature enough, and the best way to
+  debug/inspect has often been to simply open and read the file. Improving tooling for debugging
+  and visualization should help rectify this.
+
+## Why do we need program structure?
+
+There are two main reasons:
+
+- Because we *need* to (not all algorithmic primitives, an example being repeat-until-success, can
+  be represented without structure!), and
+
+- Even when we don't *need* to, preserving structure is necessary for scalability.
+
+Quantum algorithms naturally have structure, even if we aren't used to thinking of them that way.
+For example, consider Shor's algorithm: when coded up in Python, we will instinctively use
+structure such as if statements, for loops, etc., rather than writing out the finalized list of
+gates to be applied. This structure results in a *compressed* program representation compared to
+the unrolled list of quantum instructions, which is both more efficient to compile, and may even
+allow for optimizations which may not be possible when the structure is lost/the program is flattened.
+
+So ideally, we should design compiler passes to act on
+such compressed representations! In extreme cases, if the program representation does not increase in size as the program scales,
+[compilation may be constant-time](https://arxiv.org/abs/2504.12449).
+
+There is still a lot of work to do when it comes to compiling structured programs:
+
+- Often, porting known state-of-the-art compilation techniques to apply to structured programs can
+  be highly non-trivial.
+
+- There is tension between maximal optimization and compact program representation. There are still
+  a lot of questions we need to consider and resolve when it comes to compiling dynamic, structured
+  programs. For example, the more compact the program representation (which is great for
+  compilation scalability!) the less information we potentially have up-front to maximally optimize
+  programs at compile time. How do we deal with such scenarios?
+
+## How did you decide what goes into the schema?
+
+## Why Cap'n Proto?
+
+Jeff is a machine-first exchange format, so we looked at existing succh projects.
+While we could have gone with something like JSON or Messagepack - we wanted first-class
+schema support, efficient binary encoding and tooling in many languages.
+
+The [Protocol buffer](https://en.wikipedia.org/wiki/Protocol_Buffers) family of projects
+fits this description well. This includes the original 
+[Protocol buffer](https://protobuf.dev/) project but also more modern evolutions
```

Comment:
There's a lot more use of "protocol buffers" here than needed. Why not just mention "data serialization formats", or something generic, before introducing the current implementation using capnproto.

I'd reframe this as:
* We are using Capnproto
* Reasons we are using capnproto
* Alternative models to capnproto (the other models can be mentioned here)